### DISCUSSION TITLE
Can we save some I/O request via just not selecting * for just user's id?

### DISCUSSION THREAD

**Discussion by devbapidey** [laravel/framework#38613] [General]
I recently complete a sql book. Book says that we can define a BTREE index, to quickly retrieve data from index table. So we don't need to check sequentially on the table to retrieve data. Book also says sequentially check is not efficient and can take same time.

I am writing this because, when I try to get currently authenticated user id, I always use `auth()->id()`. Now this code makes a query `select * from `users` where `id` = 2 limit 1;`. 

Now there is a difference in `select * from `users` where `id` = 2 limit 1;` and `select id from `users` where `id` = 2 limit 1;`. Difference is visible when we use it like `explain select * from `users` where `id` = 2 limit 1;` and `explain select id from `users` where `id` = 2 limit 1;`

Difference is, for second query where I just need id, the sql server server search it using index.
![index](https://user-images.githubusercontent.com/33697097/131488262-41566b45-3e91-471c-804d-364926d52d3b.png)

Book says it's way much faster then search without index. Now my question is

When I just need id, then why laravel select everything in user table? Can we just use id when we just required id? @taylorotwell

**Comment by dennisprudlo** [comment]
The `auth()` helper returns an authenticator instance. When you are in an authenticated scope of your application the user is always fetched entirely. That's why you have the `select * from users...` query. You need it anyway for checking the password or setting the remember token etc.

Otherwise using `auth()->id()` and `auth()->user()` in the same request would result in two queries. One for the id and one for the whole user.
